src/dmap_import/util_api.py

```diff
@@ -47,23 +47,33 @@ def download_from_url(url: str, local_path: str) -> Optional[str]:
     download_log.log_start()
 
     max_retries = 3
-    for retry_count in range(max_retries):
+    for retry_count in range(max_retries + 1):
         download_log.add_metadata(retry_count=retry_count)
-        with requests.get(url, stream=True, timeout=60 * 5) as response:
-            # retry if status_code not 200
-            if response.status_code != 200:
-                if retry_count == max_retries - 1:
-                    response.raise_for_status()
-                # try again in 15 seconds
-                time.sleep(15)
-                continue
+        try:
+            response = requests.get(url, stream=True, timeout=15)
+            response.raise_for_status()
 
             # download file
             with open(local_path, "wb") as local_file:
                 for file_chunk in response.iter_content(chunk_size=None):
                     # chunk_size=None will write chunks in whatever size they are received
                     local_file.write(file_chunk)
-        break
+            response.close()
```

Comment:
do we need to close this response if we hit an exception in the file download?

Reply:
In my testing it hasn't caused any issues,  we probably don't really need to call it at all based on the library documentation:

```
close()[[source]](https://requests.readthedocs.io/en/latest/_modules/requests/models/#Response.close)

    Releases the connection back to the pool. Once this method has been called the underlying raw object must not be accessed again.

    Note: Should not normally need to be called explicitly.
```

Reply:
looks like at worst we'd have some connection inefficiencies if something goes wrong in the reading. in normal operation, the connection will be released after the `iter_content` for loop (when all the data is consumed).

[src](https://requests.readthedocs.io/en/latest/user/advanced/#body-content-workflow)